### PR TITLE
test: Add cases for having unicode in URL paths

### PR DESCRIPTION
**Pull request by dio**
Commit Message: Add tests of having unicode chars in URL paths

Signed-off-by: Dhi Aurrahman <dio@tetrate.io>

Additional Description:
Risk Level: Low
Testing: Test only
Docs Changes: N/A
Release Notes: N/A

### CODE REVIEW

**Comment by dio** [comment]
@asraa added the corpus. Thank you for suggesting.

File: test/common/http/utility_test.cc [hunk 1]
```diff
@@ -1289,6 +1289,10 @@ TEST(Url, ParsingTest) {
   // Test url with non-default ports.
   validateUrl("https://www.host.com:8443", "https", "www.host.com:8443", "/", 8443);
   validateUrl("http://www.host.com:8080", "http", "www.host.com:8080", "/", 8080);
+
+  // Test having unicode chars in the URL.
+  validateUrl("https://example.com/\xe5\x85\x89", "https", "example.com", "/%E5%85%89", 443);
+  validateUrl("https://example.com/光", "https", "example.com", "/%E5%85%89", 443);
```
Comment (lizan) [inline test/common/http/utility_test.cc:1295]: use `u8""` string literal.

Comment (dio) [reply test/common/http/utility_test.cc]: Hum, seems like adding "u8" fails the windows test. Where I can find the log for a particular test on CI? In this case: C:/_eb/execroot/envoy/bazel-out/x64_windows-opt/testlogs/test/common/http/utility_test/test.log?

**Comment by dio** [comment]
Seems like we don't need this one.

**Pull request closed without merging**